Messages submitted to the subscription pipeline are lost when submission fails. (#4733)
hapifhir/hapi-fhir: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_0_0/upgrade.md

> This release introduce significant changes to the mechanism performing submission of resource modification events

Reviewer: ```suggestion
This release introduces a significant change to the mechanism performing submission of resource modification events
```

> to the message broker.  The mechanism found in previous releases would submit an event as part of the transaction

Reviewer: ```suggestion
to the message broker.  Previously, an event would be submitted as part of the synchronous transaction
```

modifying a resource.  Synchronous submission yielded responsive publishing with the caveat that events would be dropped
upon submission failure.

We have replaced the synchronous mechanism with a two stage process.  Events are initially stored in

> persistence as part of the operation on a resource and subsequently submitted to the broker by a scheduled task.

Reviewer: ```suggestion
the database upon completion of the transaction, and subsequently submitted to the broker by a scheduled task.
```
